File: Fundamentele programarii (Sem1)/Curs1/Perl/vehicule_regexp.pl
#!/usr/bin/perl

# executie: perl vehicule_regexp.php

my $text = ' Lulu conduce bicicleta si masina; Judith conduce trotineta si vapor; Aladin conduce trotineta role si bicicleta; Heidi conduce avion role si bicicleta.';

print "\nCine conduce bicicleta\n";

while ($text =~ /([A-Z]\w+)\sconduce[^;]*?bicicleta.*?(;|\.)/g) {
    print $1."\n";
}

File: Fundamentele programarii (Sem1)/Curs1/Php/vehicule_regexp.php
<?php

//executie: php vehicule_regexp.php

$text = ' Lulu conduce bicicleta si masina; Judith conduce trotineta si vapor; Aladin conduce trotineta role si bicicleta; Heidi conduce avion role si bicicleta.';

$pattern = '/([A-Z]\w+)\sconduce.*?bicicleta.*?(;|\.)/';
$replacement = '$1';

$result =  preg_replace($pattern, $replacement, $text);
echo "\nCine conduce bicicleta\n";
echo $result ."\n";

